pkg/wharfyml/stage.go

```diff
@@ -23,9 +23,25 @@ type Stage struct {
 	EnvsPos visit.Pos
 	Steps   []Step
 
+	RunsIf StageRunsIf
+
 	Node visit.MapItem
 }
 
+// ShouldSkip returns true if the stage should be skipped based on its run
+// conditions.
+func (s Stage) ShouldSkip(anyPreviousStageHasFailed bool) bool {
+	switch s.RunsIf {
+	case StageRunsIfAlways:
+		return false
+	case StageRunsIfFail:
+		return !anyPreviousStageHasFailed
+	case "", StageRunsIfSuccess:
+		return anyPreviousStageHasFailed
+	}
+	return true
```

Comment:
Perhaps warn here for unhandled RunsIf value

Reply:
Good idea.
Because it errors during parsing of the wharf-ci file if it's not a valid value, I will put it as an error, and indicate in the message that it is a bug. I think that should make it easier to spot as a bug-source if it does happen.